From c7d57d5d008fdf1494a52e13a2a035e9490f7d46 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Fri, 27 Feb 2009 20:45:03 -0800
Subject: Staging: line6: static function cleanups

This fixes all of the static function warnings that sparse complains
about.

Cc: Markus Grabner <grabner@icg.tugraz.at>
Cc: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/line6/audio.c    |    1 +
 drivers/staging/line6/capture.c  |    1 +
 drivers/staging/line6/midibuf.c  |    6 +++---
 drivers/staging/line6/playback.c |    1 +
 drivers/staging/line6/pod.c      |    4 ++--
 drivers/staging/line6/variax.c   |    2 +-
 6 files changed, 9 insertions(+), 6 deletions(-)

--- a/drivers/staging/line6/audio.c
+++ b/drivers/staging/line6/audio.c
@@ -10,6 +10,7 @@
  */
 
 #include "driver.h"
+#include "audio.h"
 
 #include <sound/core.h>
 #include <sound/initval.h>
--- a/drivers/staging/line6/capture.c
+++ b/drivers/staging/line6/capture.c
@@ -18,6 +18,7 @@
 #include "audio.h"
 #include "pcm.h"
 #include "pod.h"
+#include "capture.h"
 
 
 /*
--- a/drivers/staging/line6/midibuf.c
+++ b/drivers/staging/line6/midibuf.c
@@ -16,7 +16,7 @@
 #include "midibuf.h"
 
 
-int midibuf_message_length(unsigned char code)
+static int midibuf_message_length(unsigned char code)
 {
 	if(code < 0x80)
 		return -1;
@@ -59,12 +59,12 @@ void midibuf_status(struct MidiBuffer *t
 				 this->size, this->split, this->pos_read, this->pos_write, this->full, this->command_prev);
 }
 
-int midibuf_is_empty(struct MidiBuffer *this)
+static int midibuf_is_empty(struct MidiBuffer *this)
 {
 	return (this->pos_read == this->pos_write) && !this->full;
 }
 
-int midibuf_is_full(struct MidiBuffer *this)
+static int midibuf_is_full(struct MidiBuffer *this)
 {
 	return this->full;
 }
--- a/drivers/staging/line6/playback.c
+++ b/drivers/staging/line6/playback.c
@@ -18,6 +18,7 @@
 #include "audio.h"
 #include "pcm.h"
 #include "pod.h"
+#include "playback.h"
 
 
 /*
--- a/drivers/staging/line6/pod.c
+++ b/drivers/staging/line6/pod.c
@@ -398,7 +398,7 @@ void pod_midi_postprocess(struct usb_lin
 /*
 	Send channel number (i.e., switch to a different sound).
 */
-void pod_send_channel(struct usb_line6_pod *pod, int value)
+static void pod_send_channel(struct usb_line6_pod *pod, int value)
 {
 	line6_invalidate_current(&pod->dumpreq);
 
@@ -951,7 +951,7 @@ static void pod_destruct(struct usb_inte
 /*
 	Create sysfs entries.
 */
-int pod_create_files2(struct device *dev)
+static int pod_create_files2(struct device *dev)
 {
 	int err;
 
--- a/drivers/staging/line6/variax.c
+++ b/drivers/staging/line6/variax.c
@@ -392,7 +392,7 @@ static void variax_destruct(struct usb_i
 /*
 	Create sysfs entries.
 */
-int variax_create_files2(struct device *dev)
+static int variax_create_files2(struct device *dev)
 {
 	int err;
 	CHECK_RETURN(device_create_file(dev, &dev_attr_model));
